### PR TITLE
Test sset list with pods not in sset

### PR DESCRIPTION
**Pull request by racevedoo**
This PR is simple: it only removes a TODO that said we should test sset list with pods that do not belong to the sset. It was not done before due to https://github.com/kubernetes-sigs/controller-runtime/pull/311. Relates to #1800

### CODE REVIEW

File: pkg/controller/elasticsearch/sset/list_test.go [hunk 1]
```diff
@@ -79,14 +79,20 @@ func TestStatefulSetList_GetExistingPods(t *testing.T) {
 			},
 		},
 	}
-	client := k8s.WrapClient(fake.NewFakeClient(&pod1, &pod2))
+	// pod not belonging to the sset
+	podNotInSset := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod-not-in-sset",
+			Labels: map[string]string{
+				label.StatefulSetNameLabelName: "not-a-sset",
```
Comment (sebgl) [inline pkg/controller/elasticsearch/sset/list_test.go]: maybe more `"different-sset"`

**Comment by sebgl** [comment]
Jenkins test this please.

File: pkg/controller/elasticsearch/sset/list_test.go (hunk 1 above)
Comment (racevedoo) [reply pkg/controller/elasticsearch/sset/list_test.go]: agreed, I guess I was too sleepy when I wrote this :stuck_out_tongue_closed_eyes:

**Comment by thbkrkr** [comment]
Jenkins test this please.

**Comment by racevedoo** [comment]
Hey guys, I guess jenkins got a timeout. From console output:
```
15:42:20  go: finding sigs.k8s.io v0.2.1
15:42:20  go: finding sigs.k8s.io/controller-tools/cmd v0.2.1
15:42:20  go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.1
15:42:47  verifying sigs.k8s.io/controller-tools@v0.2.1: sigs.k8s.io/controller-tools@v0.2.1: Get https://sum.golang.org/lookup/sigs.k8s.io/controller-tools@v0.2.1: dial tcp 108.177.120.141:443: i/o timeout
```

Can anyone trigger another build please?

**Comment by sebgl** [comment]
Jenkins test this please

**Pull request merged**